taui/src/utils/standardize-listings-data.js

```diff
@@ -0,0 +1,33 @@
+// @flow
+
+export default function standardizeListingsData (fn: Function): Function {
+  return function (...args) {
+    const [data, type] = args
+    let newData
+    switch (type) {
+      case 'BHA':
+        newData = {
+          photos: data.photos,
+          rent: data.Rent,
+          beds: data['Bedroom Type'] === 'Studio' ? data['Bedroom Type'] : `${data['Bedroom Type']} Bed`,
+          address: data['Apartment Number'] ? `${data.address.line} #${data['Apartment Number']}` : data.address,
+          url: data.rdc_web_url,
+          lat: data.lat,
+          lon: data.lon
+        }
+        break
+      case 'Realtor':
+        newData = {
+          photos: data.photos,
+          rent: data.price,
+          beds: `${data.beds} Bed`,
+          address: data.address.line,
+          url: data.rdc_web_url,
+          lat: data.address.lat,
+          lon: data.address.lon
+        }
+        break
+    }
+    return fn(newData)
```

Comment:
This structure of declaring an empty `newData` up front, setting it in the `switch`, then returning it at the end took me a moment to parse out, and I think it could be a little more self-contained.  Here's one possibility--keeping the switch returning from inside each case.  Or it could be done by assigning `newData` with a ternary chain (then still calling `fn()` on it at the end).

```suggestion
    switch (type) {
      case 'BHA':
        return fn({
          photos: data.photos,
          rent: data.Rent,
          beds: data['Bedroom Type'] === 'Studio' ? data['Bedroom Type'] : `${data['Bedroom Type']} Bed`,
          address: data['Apartment Number'] ? `${data.address.line} #${data['Apartment Number']}` : data.address,
          url: data.rdc_web_url,
          lat: data.lat,
          lon: data.lon
        })
      case 'Realtor':
        return fn({
          photos: data.photos,
          rent: data.price,
          beds: `${data.beds} Bed`,
          address: data.address.line,
          url: data.rdc_web_url,
          lat: data.address.lat,
          lon: data.address.lon
        })
    }
```
Actually this isn't quite the same, since if `type` doesn't match either case, this won't return anything, whereas the original will return `fn(undefined)`.  Not sure if that's an important distinction, or what the failure case of this function would ideally be.